x-pack/plugins/security_solution/public/detection_engine/rule_management_ui/components/rules_table/use_rules_table_actions.tsx

```diff
@@ -109,6 +117,29 @@ export const useRulesTableActions = ({
       },
       enabled: (rule: Rule) => !rule.immutable,
     },
+    ...(isManualRuleRunEnabled
+      ? [
+          {
+            type: 'icon',
+            'data-test-subj': 'manualRuleRunAction',
+            description: i18n.MANUAL_RULE_RUN,
+            icon: 'play',
+            name: i18n.MANUAL_RULE_RUN,
+            onClick: async (rule: Rule) => {
+              startTransaction({ name: SINGLE_RULE_ACTIONS.MANUAL_RULE_RUN });
+              const modalManualRuleRunConfirmationResult = await showManualRuleRunConfirmation();
+              if (modalManualRuleRunConfirmationResult === null) {
+                return;
+              }
+              await scheduleRuleRun({
+                ruleIds: [rule.id],
+                timeRange: modalManualRuleRunConfirmationResult,
+              });
+            },
+            enabled: (rule: Rule) => rule.enabled,
```

Comment:
you are checking for user permissions on details page: `x-pack/plugins/security_solution/public/detections/components/rules/rule_actions_overflow/index.tsx`

but not here. Should user permissions check also be added here?

Reply:
It is already checked in [the parent component](https://github.com/elastic/kibana/blob/main/x-pack/plugins/security_solution/public/detection_engine/rule_management_ui/components/rules_table/use_columns.tsx#L346), that's why no permissions checks are done inside this hook